src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -407,8 +440,28 @@ static URLConnection openURLConnection(URL url, String credentials, String proxy
             String encoding = Base64.encode(proxyCredentials.getBytes("UTF-8"));
             con.setRequestProperty("Proxy-Authorization", "Basic " + encoding);
         }
-        if (con instanceof HttpsURLConnection && sslSocketFactory != null) {
-            ((HttpsURLConnection) con).setSSLSocketFactory(sslSocketFactory);
+
+        if (con instanceof HttpsURLConnection) {
+            final HttpsURLConnection httpsConnection = (HttpsURLConnection) con;
+            if (insecure) {
+                System.out.println(String.format("Insecure Status: %s", insecure));
```

Comment:
The use of LOGGER.fine / info should be preferred no ?

Reply:
I think so, but the file is full of stdout/stderr. @MattLud has followed the current implementation. I think it makes sense to stick to it for a while, but I will create a follow-up ticket to rework it